types/dec_coin.go

```diff
@@ -179,7 +179,9 @@ func (coins DecCoins) TruncateDecimal() (Coins, DecCoins) {
 	for i, coin := range coins {
 		truncated, change := coin.TruncateDecimal()
 		out[i] = truncated
-		changeSum = changeSum.Add(DecCoins{change})
+		if !change.IsZero() {
+			changeSum = changeSum.Add(DecCoins{change})
+		}
```

Comment:
you mention in your PR comment
> Note, we should probably update and fix TruncateDecimal to not even return zero coins 

is that not what's being done here?

Reply:
Yes -- remnants. Removed from the PR body.